File: ports/ts/src/conform/fixtures.ts
// Loads the shared cross-language conformance fixtures at <repo>/conformance/
// (SPEC-PORTS-v1 §10). These fixtures ARE the port contract: every §4
// deterministic algorithm is proven identical to the Python reference by making
// them pass, byte-for-byte.
import { readFileSync } from "node:fs";
import { fileURLToPath } from "node:url";
import { dirname, join } from "node:path";

const here = dirname(fileURLToPath(import.meta.url));
// here = <repo>/ports/ts/src/conform → up 4 to <repo>.
const conformanceDir = join(here, "..", "..", "..", "..", "conformance");

/** Parse conformance/cases/<name> (e.g. "tokenize.json"). */
export function loadCase<T>(name: string): T {
  return JSON.parse(readFileSync(join(conformanceDir, "cases", name), "utf8")) as T;
}

/** Parse a top-level conformance file (e.g. "stopwords.json"). */
export function loadData<T>(name: string): T {
  return JSON.parse(readFileSync(join(conformanceDir, name), "utf8")) as T;
}

File: python/tests/conformance/fixtures.py
"""Loader for the committed cross-language conformance fixtures.

Mirrors ``golang/internal/conform`` and ``js/src/conform/fixtures.ts``: the
fixture is read from disk as opaque JSON, never regenerated. If a port helper
exists, Python gets one too — otherwise every Python test re-derives the path
and the "read it as data" discipline erodes one copy-paste at a time.
"""

from __future__ import annotations

import json
from pathlib import Path
from typing import Any

CONFORMANCE_DIR = Path(__file__).resolve().parents[3] / "conformance"


def load_case(name: str) -> Any:
    """Parse ``conformance/cases/<name>`` (e.g. ``"tokenize.json"``)."""
    return json.loads((CONFORMANCE_DIR / "cases" / name).read_text(encoding="utf-8"))


def load_data(name: str) -> Any:
    """Parse a top-level ``conformance/<name>`` (e.g. ``"stopwords.json"``)."""
    return json.loads((CONFORMANCE_DIR / name).read_text(encoding="utf-8"))
